drivers/dma/dma_andes_atcdmac300.h

```diff
@@ -0,0 +1,165 @@
+#define DMA_WIDTH_WORD (2)
+#define DMA_WIDTH_DWORD (3)
+#define DMA_WIDTH_QWORD (4)
+#define DMA_WIDTH_EWORD (5)
+
+/* Bus interface index */
+#define DMA_INF_IDX0 (0)
+#define DMA_INF_IDX1 (1)
+
+/* DMA Channel Control Register Definition */
+#define DMA_CH_CTRL_SBINFIDX_POS (31)
+#define DMA_CH_CTRL_SBINF(n) ((n) << DMA_CH_CTRL_SBINFIDX_POS)
+#define DMA_CH_CTRL_SBINF_MASK (1 << DMA_CH_CTRL_SBINFIDX_POS)
+#define DMA_CH_CTRL_DBINFIDX_POS (30)
+#define DMA_CH_CTRL_DBINF(n) ((n) << DMA_CH_CTRL_DBINFIDX_POS)
+#define DMA_CH_CTRL_DBINF_MASK (1 << DMA_CH_CTRL_DBINFIDX_POS)
+#define DMA_CH_CTRL_PRIORITY_HIGH (1 << 29)
+#define DMA_CH_CTRL_SBSIZE_POS (24)
+#define DMA_CH_CTRL_SBSIZE_MASK (0x0f << DMA_CH_CTRL_SBSIZE_POS)
+#define DMA_CH_CTRL_SBSIZE(n) \
+	(((n) << DMA_CH_CTRL_SBSIZE_POS) & DMA_CH_CTRL_SBSIZE_MASK)
+#define DMA_CH_CTRL_SWIDTH_POS (21)
+#define DMA_CH_CTRL_SWIDTH_MASK (0x07 << DMA_CH_CTRL_SWIDTH_POS)
+#define DMA_CH_CTRL_SWIDTH(n) \
+	(((n) << DMA_CH_CTRL_SWIDTH_POS) & DMA_CH_CTRL_SWIDTH_MASK)
+#define DMA_CH_CTRL_DWIDTH_POS (18)
+#define DMA_CH_CTRL_DWIDTH_MASK (0x07 << DMA_CH_CTRL_DWIDTH_POS)
+#define DMA_CH_CTRL_DWIDTH(n) \
+	(((n) << DMA_CH_CTRL_DWIDTH_POS) & DMA_CH_CTRL_DWIDTH_MASK)
+#define DMA_CH_CTRL_SMODE_HANDSHAKE (1 << 17)
+#define DMA_CH_CTRL_DMODE_HANDSHAKE (1 << 16)
+#define DMA_CH_CTRL_SRCADDRCTRL_POS (14)
+#define DMA_CH_CTRL_SRCADDRCTRL_MASK (0x03 << DMA_CH_CTRL_SRCADDRCTRL_POS)
+#define DMA_CH_CTRL_SRCADDR_INC (0 << DMA_CH_CTRL_SRCADDRCTRL_POS)
+#define DMA_CH_CTRL_SRCADDR_DEC (1 << DMA_CH_CTRL_SRCADDRCTRL_POS)
+#define DMA_CH_CTRL_SRCADDR_FIX (2 << DMA_CH_CTRL_SRCADDRCTRL_POS)
+#define DMA_CH_CTRL_DSTADDRCTRL_POS (12)
+#define DMA_CH_CTRL_DSTADDRCTRL_MASK (0x03 << DMA_CH_CTRL_DSTADDRCTRL_POS)
+#define DMA_CH_CTRL_DSTADDR_INC (0 << DMA_CH_CTRL_DSTADDRCTRL_POS)
+#define DMA_CH_CTRL_DSTADDR_DEC (1 << DMA_CH_CTRL_DSTADDRCTRL_POS)
+#define DMA_CH_CTRL_DSTADDR_FIX (2 << DMA_CH_CTRL_DSTADDRCTRL_POS)
+#define DMA_CH_CTRL_SRCREQ_POS (8)
+#define DMA_CH_CTRL_SRCREQ_MASK (0x0F << DMA_CH_CTRL_SRCREQ_POS)
+#define DMA_CH_CTRL_SRCREQ(n) \
+	(((n) << DMA_CH_CTRL_SRCREQ_POS) & DMA_CH_CTRL_SRCREQ_MASK)
+#define DMA_CH_CTRL_DSTREQ_POS (4)
+#define DMA_CH_CTRL_DSTREQ_MASK (0x0F << DMA_CH_CTRL_DSTREQ_POS)
+#define DMA_CH_CTRL_DSTREQ(n) \
+	(((n) << DMA_CH_CTRL_DSTREQ_POS) & DMA_CH_CTRL_DSTREQ_MASK)
+#define DMA_CH_CTRL_INTABT (1 << 3)
+#define DMA_CH_CTRL_INTERR (1 << 2)
+#define DMA_CH_CTRL_INTTC (1 << 1)
+#define DMA_CH_CTRL_ENABLE (1 << 0)
+
+/* DMA Interrupt Status Register Definition */
+#define DMA_INT_STATUS_TC_POS (16)
+#define DMA_INT_STATUS_ABORT_POS (8)
+#define DMA_INT_STATUS_ERROR_POS (0)
+#define DMA_INT_STATUS_TC_VAL(x) ((x >> DMA_INT_STATUS_TC_POS) & 0xff)
+#define DMA_INT_STATUS_ABORT_VAL(x) ((x >> DMA_INT_STATUS_ABORT_POS) & 0xff)
+#define DMA_INT_STATUS_ERROR_VAL(x) (x & 0xff)
+#define DMA_INT_STATUS_CH_MSK(ch) (0x111 << ch)
+
+#define INWORD(x) sys_read32(x)
+#define OUTWORD(x, d) sys_write32(d, x)
```

Comment:
Why redefined these?

Reply:
Thanks for your remind, these macro is redundant.
We have update the source code for it.